### ISSUE TITLE
"Invalid non-string/buffer chunk" when piping

### ISSUE THREAD

**Issue by ivan** [calvinmetcalf/chacha20poly1305#6]
If you set up a `.pipe()` to encrypt stdin:

```
var chacha = require('chacha');

var password = Buffer('3zTvzr3p67VC61jmV54rIYu1545x4TlY');
var nonce = Buffer('abcdef123456');

var cipher = chacha.createCipher(password, nonce);
process.stdin.pipe(cipher).pipe(process.stdout);
```

and send a file through it, you'll get a TypeError after the end of the file:

```
# cat /etc/passwd | iojs index.js > /dev/null 
events.js:141
      throw er; // Unhandled 'error' event
            ^
TypeError: Invalid non-string/buffer chunk
    at chunkInvalid (_stream_readable.js:367:10)
    at readableAddChunk (_stream_readable.js:123:12)
    at AEAD.Readable.push (_stream_readable.js:109:10)
    at AEAD.Transform.push (_stream_transform.js:128:32)
    at AEAD.CipherBase._flush (/mnt/devdrive/NodeProjects/cryptocompare/node_modules/chacha/node_modules/chacha-native/lib/cipherbase.js:55:10)
    at AEAD.<anonymous> (_stream_transform.js:118:12)
    at AEAD.g (events.js:257:16)
    at emitNone (events.js:67:13)
    at AEAD.emit (events.js:163:7)
    at prefinish (_stream_writable.js:463:12)
```

**Comment by calvinmetcalf** [comment]
fixed in chacha-native 1.2.1 do a `rm -rf node_modules && npm install` and it should be fixed

**Comment by ivan** [comment]
Thanks! That works now.
